platform/bukkit/plugin/src/main/java/me/lokka30/treasury/plugin/bukkit/hooks/EconomyHook.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * This file is/was part of Treasury. To read more information about Treasury such as its licensing, see <https://github.com/lokka30/Treasury>.
+ */
+
+package me.lokka30.treasury.plugin.bukkit.hooks;
+
+import java.math.BigDecimal;
+import java.text.DecimalFormat;
+import java.util.Optional;
+import me.lokka30.treasury.api.common.service.Service;
+import me.lokka30.treasury.api.common.service.ServiceRegistry;
+import me.lokka30.treasury.api.economy.EconomyProvider;
+import me.lokka30.treasury.api.economy.account.PlayerAccount;
+import me.lokka30.treasury.api.economy.response.EconomySubscriber;
+import org.bukkit.OfflinePlayer;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public class EconomyHook implements TreasuryPAPIHook {
+
+    private EconomyProvider provider;
+    private final DecimalFormat format = new DecimalFormat("#,###");
```

Comment:
Should we use `#,##0.00`?

Reply:
It is 0,00 currently. We can change it to 0.00

Reply:
From memory, using the DF I suggested, it will look like this:
* `1.2345` = `1.23`
* `25.3` = `25.30`
* `1234` = `1,234.00`
* `1234567.89` = `1,234,567.89` (unsure if there will be two commas or only one)

Reply:
> It is 0,00 currently. We can change it to 0.00

The period is just a decimal separator notation and the comma is a grouping separator notation specifically for declaring the format. You can force a specific locale's symbols by specifying the `DecimalFormatSymbols` to be used, but otherwise they will be localized by the formatter for your system locale. 

> From memory, using the DF I suggested, it will look like this:
> * `1234567.89` = `1,234,567.89` (unsure if there will be two commas or only one)

There will be two. The number of characters between the last grouping separator and the decimal separator (or the end of the pattern) is applied to any additional multiples of the grouping separator. A pattern `#,##,###` applied to `100000` should actually yield `100,000` and not `1,00,000`.

Reply:
Unless you don't like it, perhaps we could ask the PAPI team if it's a better decimal format they could use for their Vault one as well

Reply:
Even though the proposed DecimalFormat is better, let's keep it the way PAPI has it